src/librustdoc/html/render/mod.rs

```diff
@@ -1674,19 +1670,12 @@ pub(crate) fn render_impl_summary(
     containing_item: &clean::Item,
     show_def_docs: bool,
     use_absolute: Option<bool>,
-    is_on_foreign_type: bool,
     // This argument is used to reference same type with different paths to avoid duplication
     // in documentation pages for trait with automatic implementations like "Send" and "Sync".
     aliases: &[String],
 ) {
     let id = cx.derive_id(match i.inner_impl().trait_ {
-        Some(ref t) => {
-            if is_on_foreign_type {
-                get_id_for_impl_on_foreign_type(&i.inner_impl().for_, t, cx)
-            } else {
-                format!("impl-{}", small_url_encode(format!("{:#}", t.print(cx))))
-            }
-        }
+        Some(ref t) => get_id_for_impl(&i.inner_impl().for_, Some(t), cx),
         None => "impl".to_string(),
```

Comment:
> This test is still passing when it should fail:

@pierwill you forgot to update this line ;)

Reply:
You can just get rid of the whole match and pass `i.inner_impl().trait_.as_ref()` directly as the argument to `get_id_for_impl`.

Reply:
I don't get it 😫 lol

Reply:
Current:

```rust
    let id = cx.derive_id(match i.inner_impl().trait_ {
        Some(ref t) => get_id_for_impl(&i.inner_impl().for_, Some(t), cx),
        None => "impl".to_string(),
    });
```

What it needs to be:

```rust
    let id = cx.derive_id(
        get_id_for_impl(&i.inner_impl().for_, i.inner_impl().trait_.as_ref(), cx),
    });
```